### DISCUSSION TITLE
Planned changes to the Home Assistant integration - deprecating the forecast sensor.

### DISCUSSION THREAD

**Discussion by madpilot** [amberelectric/public-api#227] [Announcements]
As we move to 5-minute settlement, and with changes to how Home Assistant does things, there are a few upcoming changes that I want to socialise.

I'll be deprecating the forecast sensor soon, and replacing it with a forecast service. Home assistant have stated that stuffing reams of data into attributes is no longer a supported way of doing things, and that it's unlikely that https://github.com/home-assistant/core/pull/124008 will get approved if we add _additional_ data to the forecast sensor. Their recommendation it to create a service that you can call to receive the data.

This actually makes more sense, and gives you a bit more flexibility, as you can request as many, or as few data points as your automation needs. It also avoids the need to have switchable resolutions in HASS (as it currently stands, the 5 minute settlement change will cause the forecast horizon to reduce from ~16 hours to ~2 hours).

As such, I won't be adding the advanced price forecasts to the forecast sensor - they will only be available in the service.

As is custom with HASS, the forecast sensor will stick around for 6 months from the point the deprecation gets announced, so you'll have plenty of time to update your automations to consume the service.

**Comment by sentience** [comment]
This will be so much better! I'm excited to delete the current inscrutable code I use to parse the attribute value. :)

Thanks for letting us know!

**Comment by paulhomes** [comment]
Do I understand you correctly when you say that "_as it currently stands, the 5 minute settlement change will cause the forecast horizon to reduce from ~16 hours to ~2 hours_" is only if you were to continue to use a forecast sensor and keep the same number of forecast data points (i.e. what this change intends to avoid)?  I assume the new forecast service will allow access to forecast data beyond a 2 hour horizon with 5MS?

Thanks
Paul

**Reply by madpilot** [reply]
Yeah, you'll be able to explicitly set the `next` parameter which controls how many intervals are returned. 

The current forecast sensor has it hard coded to `48`, which is up to 24 hours (but usually 16 depending on how many forecasts are _actually_ available) in the current world, but will only be 4 hours in the new world (clearly, I couldn't maths yesterday).

So, if you wanted 24 hours worth of 5-minute forecasts from the new service, you'd need to set the `next` param to `144`

**Reply by paulhomes** [reply]
That's great. Thanks!

**Reply by CuongHome** [reply]
could you give an example of this

**Comment by altonius** [comment]
thanks for letting us know, I've been following the conversation over in the HA Github repo of the "friendly suggestion" to move to a service.
will be happy to be an early tester when the service is available.
